### PR TITLE
Update stapler to 1.262

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

See https://github.com/stapler/stapler/pull/197

### Proposed changelog entries

* Entry 1: Update stapler to 1.262, see [CHANGELOG](https://github.com/stapler/stapler/releases/tag/stapler-parent-1.262), fixes a number of IllegalReflectiveAccessWarnings when running on Java 11
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: bom/pom.xml
Comment (timja) [inline bom/pom.xml:44]:
```suggestion
    <stapler.version>1.262-rc1463.b4530fe9d42f</stapler.version>
```

File: bom/pom.xml
Comment (jglick) [inline bom/pom.xml:44]:
```suggestion
    <stapler.version>1.262-rc1463.b4530fe9d42f</stapler.version> <!-- TODO https://github.com/stapler/stapler/pull/197 -->
```

File: bom/pom.xml
Comment (timja) [inline bom/pom.xml:44]:
```suggestion
    <stapler.version>1.262-rc1475.5f6120138f06</stapler.version> <!-- TODO https://github.com/stapler/stapler/pull/201 -->
```

File: bom/pom.xml
Comment (timja) [inline bom/pom.xml:44]:
```suggestion
    <stapler.version>1.262</stapler.version>
```

**Comment by daniel-beck** [comment]
https://github.com/jenkins-infra/jenkins.io/pull/4026#issuecomment-745850851 indicates this might be quite noisy. I had hoped the code path is only encountered when these methods are actually called, but that doesn't seem to be the case.

**Comment by timja** [comment]
> [jenkins-infra/jenkins.io#4026 (comment)](https://github.com/jenkins-infra/jenkins.io/pull/4026#issuecomment-745850851) indicates this might be quite noisy. I had hoped the code path is only encountered when these methods are actually called, but that doesn't seem to be the case.

I fixed those issues in a follow up PR in stapler which is merged into this release, https://github.com/stapler/stapler/pull/201

I didn't encounter any issues since that fix, but any additional interactive testing would be welcomed.

**Review by oleg-nenashev — APPROVED** [review]
Upstream change LGTM

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

**Pull request merged**
